### PR TITLE
Support deep link to paginated replies

### LINKED ISSUE #738: [Bug]: Deep link replies that are in nested "pages" on the reply thread
### What happened?

Currently if you try to deep link a reply that is not on the initial page it will not find it

### What browsers are you seeing the problem on?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by ayrock-dev**
## Description

 This PR includes the following changes:

 * Update pagination of replies to use a cursor in the `useInfiniteQuery` for replies.
   * This enables increasing the size of the first "page" of replies (replies above the "Load More") to include the deep-linked reply.
 * Adjust `scrollIntoView()` to use the `"nearest"` edge option ([reference](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView#block)).
   * This prevents a bug where the comment container is shifted "up" when the DOM attempts to scroll the preselected reply into view on first paint. (We can iterate here I'm not in love with it, but as-is this fixes the bug on `main`)
* Adjusted background color of selected page button ([1], [2], [3]) so that it color matches the selected tab in the head of the sidebar ("Description", "Solutions", etc).
  * This was suggested by @PickleNik.
* Replies are now in newest-to-oldest order from server. Clicking "Load More" loads in the next 5 older replies

## Related Issue

Closes #738

## Motivation and Context

We want to be able to deep link to a reply (which is a comment on a comment), and scroll to it on page load when someone visits the deep link.

## How Has This Been Tested?

Deep links to replies on non-first pages tested in video below. Also tested the new page button background color in dark mode and light mode at the end of the video.

**Test Steps**

- Add comments to a **Challenge**, so that there are >1 pages
- Add replies to a comment not on the first page, so that there are also >1 pages of replies on a comment
- Click share on one of those replies
- Hard navigate your browser to the share URL

Assertions

- [x] Window scrolls the reply into view
- [x] You do not have to click "Load More" to see the reply

The above test case can be repeated for comments on a **Solution**

## Video

https://github.com/typehero/typehero/assets/7817695/f2b16a99-7f21-49ed-bf2a-89a8bd932c04

### CODE REVIEW

File: apps/web/src/app/[locale]/challenge/_components/comments/index.tsx [hunk 1]
```diff
@@ -256,7 +256,9 @@ function Pagination({
       </Button>
       {pages.map((page) => (
         <Button
-          className={clsx({ 'border border-black dark:border-white/30 ': page === currentPage })}
+          className={clsx('border-border dark:border-ring border', {
+            'bg-border dark:bg-neutral-700': page === currentPage,
+          })}
```
Comment (ayrock-dev) [inline apps/web/src/app/[locale]/challenge/_components/comments/index.tsx:261]: color match suggestion by @PickleNik 

| Light | Dark |
|----|----|
|Screenshot 2023-09-30 at 10 48 29 PM|Screenshot 2023-09-30 at 10 51 03 PM|

File: apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx [hunk 2]
```diff
@@ -198,34 +212,35 @@ export function Comment({
...
-                comment={reply}
-                isReply
-                key={reply.id}
-                replyQueryKey={replyQueryKey}
-                preselectedCommentMetadata={preselectedCommentMetadata}
-              />
-            )),
-          )}
-        </div>
+        <>
```
Comment (ayrock-dev) [inline apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx:217]: "Load More" button moved beneath replies. Clicking "Load More" loads older replies.

| Dark | Light |
| ---- | ---- |
|Screenshot 2023-09-30 at 10 54 27 PM|Screenshot 2023-09-30 at 10 53 44 PM|

File: apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx [hunk 3]
```diff
@@ -75,6 +75,8 @@ const commentReportSchema = z
 
 export type CommentReportSchemaType = z.infer<typeof commentReportSchema>;
 
+const REPLIES_PAGESIZE = 5;
+
```
Comment (ayrock-dev) [inline apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx:79]: Adjusted replies to show just 5 at a time, this matches LeetCode

For example, https://leetcode.com/problems/two-sum/description/comments/1565865 
image

File: apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx [hunk 4]
```diff
@@ -198,34 +212,35 @@ export function Comment({
...
-                isReply
-                key={reply.id}
-                replyQueryKey={replyQueryKey}
-                preselectedCommentMetadata={preselectedCommentMetadata}
-              />
-            )),
-          )}
-        </div>
+        <>
+          <div className="flex flex-col gap-1 pl-6 pt-1">
```
Comment (ayrock-dev) [inline apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx:218]: I swapped `flex-col-reverse` for `flex-col`. This puts newest replies on top, older replies going down. Pressing "Load More" gets the next page of older replies. This matches LeetCode

For example, https://leetcode.com/problems/two-sum/description/comments/1565865
image

Comment (bautistaaa) [reply apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx:218]: yeah we had a mix of linkedin behavior and leetcode

linked has it reversed. i dont have a pref here so im okay with this but we can iterate if need be

**Review by bautistaaa — APPROVED** [review]
hell yeah thanks dude!

File: apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx (hunk 4 above)
Comment (ayrock-dev) [reply apps/web/src/app/[locale]/challenge/_components/comments/comment.tsx:218]: Thanks for the background, I'm around if we change our mind :+1:

**Pull request merged**